frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BaseCheckoutAction.java

```diff
@@ -0,0 +1,131 @@
+import com.virtuslab.gitmachete.frontend.resourcebundles.GitMacheteBundle;
+import com.virtuslab.gitmachete.frontend.vfsutils.GitVfsUtils;
+import com.virtuslab.qual.guieffect.UIThreadUnsafe;
+
+@ExtensionMethod({GitVfsUtils.class, GitMacheteBundle.class})
+@CustomLog
+public abstract class BaseCheckoutAction extends BaseGitMacheteRepositoryReadyAction
+    implements
+      IExpectsKeySelectedBranchName {
+
+  @Override
+  public LambdaLogger log() {
+    return LOG;
+  }
+
+  protected abstract @Nullable String getTargetBranchName(AnActionEvent anActionEvent);
+
+  @Override
+  @UIEffect
+  protected void onUpdate(AnActionEvent anActionEvent) {
+    super.onUpdate(anActionEvent);
+
+    val presentation = anActionEvent.getPresentation();
+    if (!presentation.isEnabledAndVisible()) {
+      return;
+    }
+
+    val targetBranchName = getTargetBranchName(anActionEvent);
+
+    // It's very unlikely that targetBranchName is empty at this point since it's assigned directly before invoking this
+    // action in EnhancedGraphTable.EnhancedGraphTableMouseAdapter#mouseClicked; still, it's better to be safe.
+    if (targetBranchName == null || targetBranchName.isEmpty()) {
+      presentation.setEnabled(false);
+      presentation.setDescription(getNonHtmlString("action.GitMachete.BaseCheckoutAction.undefined.branch-name"));
+      return;
+    }
+
+    val currentBranchName = getCurrentBranchNameIfManaged(anActionEvent);
+
+    if (currentBranchName != null && currentBranchName.equals(targetBranchName)) {
+      presentation.setEnabled(false);
+      presentation.setDescription(
+          getNonHtmlString("action.GitMachete.CheckoutSelectedAction.description.disabled.currently-checked-out")
+              .format(targetBranchName));
+
+    } else {
+      presentation.setDescription(
+          getNonHtmlString("action.GitMachete.CheckoutSelectedAction.description.precise")
+              .format(targetBranchName));
+    }
+  }
+
+  @Override
+  @UIEffect
+  public void actionPerformed(AnActionEvent anActionEvent) {
+    val targetBranchName = getTargetBranchName(anActionEvent);
+    if (targetBranchName == null || targetBranchName.isEmpty()) {
+      return;
+    }
+
+    val project = getProject(anActionEvent);
+    val gitRepository = getSelectedGitRepository(anActionEvent);
+
+    if (gitRepository != null) {
+
+      log().debug(() -> "Queuing '${targetBranchName}' branch checkout background task");
+      new Task.Backgroundable(project, getString("action.GitMachete.CheckoutSelectedAction.task-title")) {
+        @Override
+        @UIThreadUnsafe
+        public void run(ProgressIndicator indicator) {
+          doCheckout(project, indicator, targetBranchName, gitRepository);
+        }
+      }.queue();
+    }
+  }
+
+  @UIThreadUnsafe
+  public static void doCheckout(Project project, @NonNull ProgressIndicator indicator, String branchToCheckoutName,
```

Comment:
`@NonNull` - do we need it?

Reply:
I guess this was the preexisting code, I refactored from our `CheckoutSelectedAction`. I would try to see if the `@NonNull` is still needed.

Reply:
Almost surely not needed, non-nullness is always assumed by default by Checker